### PR TITLE
Add support to the multiple file format support feature. 

### PR DESCRIPTION
**Pull request by kylasa**
This Pull Request is part of the Distributed Graph Partitioning feature pipeline to support massively large graphs for distributed training. This is to add support to the ParMETIS to read input graph files which are in multiple file format. Here it is assumed that each process is mapped to a input graph file for each of the graph files (nodes, edges, node features and edge features). Each process reads nodes/edges file which is assigned to the current process, using the process-id as suffix string to the appropriate input graph file name. After reading the input graph file, we as

### CODE REVIEW

File: programs/dglpart2.c [hunk 1]
```diff
@@ -251,44 +249,79 @@ graph_t *DGLPart_ReadGraph(char *fstem, MPI_Comm comm)
...
     edge_t **coo_buffers=NULL, **coo_chunks=NULL, *lcoo=NULL;
+    edge_t *snd_buffer=NULL, *rcv_buffer=NULL;
     idx_t firstvtx, lastvtx;
+    idx_t mflinesread;
+    idx_t *rcv_buffer_count=NULL;
+    idx_t *snd_displs=NULL, *rcv_displs=NULL;
+    char *newstr, *curstr;
 
     chunksize = CHUNKSIZE;
     nchunks = 1+gnedges/chunksize;
```
Comment (Rhett-Ying) [inline programs/dglpart2.c:263]: if `gnedges%chunksize == 0`, does `+1` work as expected? same issue for below `edgecount`.

File: programs/CMakeLists.txt [hunk 2]
```diff
@@ -7,11 +7,12 @@ add_executable(pm_parmetis parmetis.c io.c adaptgraph.c)
 add_executable(pm_pometis pometis.c io.c)
 add_executable(pm_dglpart dglpart.c)
 add_executable(pm_dglpart2 dglpart2.c)
+add_executable(pm_dglpart3 dglpart3.c)
```
Comment (Rhett-Ying) [inline programs/CMakeLists.txt:10]: where's `dglpart3.c`? the below file should be renamed?

File: programs/dglpart2.c [hunk 3]
```diff
@@ -55,15 +54,15 @@ int main(int argc, char *argv[])
   gkMPI_Comm_size(comm, &npes);
   gkMPI_Comm_rank(comm, &mype);
 
-  if (argc != 3) {
+  if (argc != 5) {
     if (mype == 0)
-      printf("Usage: %s <fstem> <nparts>\n", argv[0]);
+      printf("Usage: %s <fstem> <nparts> <nodes_file> <edges_file>\n", argv[0]);
```
[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply programs/dglpart2.c:59]: after reviewing https://github.com/dmlc/dgl/pull/4529, the original `xxx_nodes.txt` and `xxx_edges.txt` have been replaced by `nodes_file` and `edges_files` which includes the all `xxx_nodes/edges.txt` file path. right? If yes, do we need 5 arguments here?

File: programs/dglpart2.c [hunk 4]
```diff
@@ -176,26 +176,24 @@ graph_t *DGLPart_ReadGraph(char *fstem, MPI_Comm comm)
...
+    printf("ERROR: File '%s' does not exists.\n", filename);
+    ier++;
+  }
 
-    sprintf(filename, "%s_nodes.txt", fstem);
-    if (!gk_fexists(filename)) {
-      printf("ERROR: File '%s' does not exists.\n", filename);
-      ier++;
-    }
+  //sprintf(filename, "%s_nodes_%02d.txt", fstem, mype);
```
Comment (Rhett-Ying) [inline programs/dglpart2.c:193]: I think the content(namely the real path of `xxx_nodes/edges.txt`) in these 2 files should be checked here.

**Review by Rhett-Ying — CHANGES_REQUESTED** [review]

File: programs/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (kylasa) [reply programs/CMakeLists.txt:10]: That is the convension which is being followed in this code base. I guess this is something which we cannot change right now.

File: programs/dglpart2.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (kylasa) [reply programs/dglpart2.c]: No. I think there is some mis-understanding here. I added only two arguments `nfiles` and `efiles` paramters to the existing command. So I guess we only need 4 parameters.

[the remark this answers is not in the document]
Comment (kylasa) [reply programs/dglpart2.c]: About ReadME and example usage, please propose a place where this sample usage command can be placed. But I believe our dgl.doc.ai documentation should have enough documentation about the way we plan to use ParMETIS.

File: programs/CMakeLists.txt (hunk 2 above)
Comment (kylasa) [reply programs/CMakeLists.txt:10]: dglpart3.c exits in the `programs' folder of the ParMETIS installation.

File: programs/dglpart2.c (hunk 4 above)
Comment (kylasa) [reply programs/dglpart2.c]: Done.

**Pull request merged**